py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,755 @@
+# -*- coding: utf-8 -*-
+'''
+Ultimately customizable weather module based on the Timezone API
+(https://timezoneapi.io) and the OpenWeatherMap API
+(https://openweathermap.org).
+
+Requires an API key for OpenWeatherMap (OWM), but the free tier allows you
+enough requests/sec to get accurate weather even up to the minute.
+
+This module allows you to specify an icon for nearly every weather scenario
+imaginable. The default configuration options lump many of the icons into
+a few groups, and due to the limitations of UTF-8, this is really as expressive
+as it gets.
+
+I would highly suggest you install an additional font, such as the incredible
+(and free!) Weather Icons font (https://erikflowers.github.io/weather-icons),
+which has icons for most weather scenarios. But, this will still work with
+the i3bar default font, Deja Vu Sans Mono font, which has Unicode support.
+You can see the (limited) weather icon support within Unicode in the defaults.
+
+For more information, see the documentation
+(https://openweathermap.org/weather-conditions) on what weather conditions are
+supported. See the configuration options for how to specify each weather icon.
+
+Configuration parameters:
+    api_key: Your OpenWeatherMap API key
+        See https://openweathermap.org/appid. Required!
+        (default None)
+    cache_timeout: The time between weather polling in seconds
+        It is recommended to keep this at a higher value to avoid rate
+        limiting with the API's.
+        (default 600)
+    forecast_days: Number of days to include in the forecast, including today
+        (regardless of the 'forecast_include_today' flag)
+        (default 3)
+    forecast_include_today: Include today in the forecast? (Boolean)
+        (default False)
+    forecast_text_separator: Separator between entries in the forecast
+        (default ' ')
+    format: How to display the weather
+        This also dictates the type of forecast. The placeholders here refer to
+        the format_[...] variables found below.
+        Available placeholders:
+            icon, city, clouds, rain, snow, wind, humidity, pressure,
+            temperature, sunrise, sunset, main, description, forecast
+        (default '{city} {icon} {temperature}{rain}, {description} {forecast}')
```

Comment:
(Fix me) `format_clouds` too. I'm currently at `1%` right now wondering if it'll reach `0%`.